<a href="https://colab.research.google.com/github/LRisaro/TP_DATOS_2021/blob/main/TP_DATOS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
from pandas_profiling import ProfileReport
from matplotlib import pyplot as plt
import seaborn as sbn
from sklearn import tree
from sklearn.metrics import accuracy_score

%matplotlib inline

plt.style.use('default') 

sbn.set(style="whitegrid") 

In [2]:
def GetData():
  return pd.read_csv("https://docs.google.com/spreadsheets/d/1-DWTP8uwVS-dZY402-dm0F9ICw_6PNqDGLmH0u8Eqa0/export?format=csv&gid=2054774482")

**Inspección inicial de datos**

In [3]:
df = GetData()
df

anios_estudiados  ...                    trabajo
0                    17  ...            entretenimiento
1                    17  ...          directivo_gerente
2                    13  ...                  limpiador
3                    11  ...                  limpiador
4                    17  ...  profesional_especializado
...                 ...  ...                        ...
32556                16  ...            soporte_tecnico
32557                13  ...                  inspector
32558                13  ...            entretenimiento
32559                13  ...            entretenimiento
32560                13  ...          directivo_gerente

[32561 rows x 13 columns]

In [4]:
# Inspeccionamos las dimensiones iniciales del dataframe
df.shape

(32561, 13)

In [5]:
# Verifico que "tiene_alto_valor_adquisitivo" no tenga valores nulos
df['tiene_alto_valor_adquisitivo'].isnull().sum()

0

In [6]:
# Verificamos si hay campos con valores nulos
df.isnull().sum()

anios_estudiados                                 0
barrio                                         583
categoria_de_trabajo                          1836
edad                                             0
educacion_alcanzada                              0
estado_marital                                   0
ganancia_perdida_declarada_bolsa_argentina       0
genero                                           0
horas_trabajo_registradas                        0
religion                                         0
rol_familiar_registrado                          0
tiene_alto_valor_adquisitivo                     0
trabajo                                       1843
dtype: int64

**Análisis de columnas**

**Años estudiados**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?

**Análisis**

In [16]:
# ¿Tiene valores nulos?
df["anios_estudiados"].isnull().sum()

0

In [17]:
df["anios_estudiados"].value_counts()

13    10501
14     7291
17     5355
18     1723
15     1382
11     1175
16     1067
10      933
8       646
19      576
9       514
12      433
20      413
6       333
4       168
1        51
Name: anios_estudiados, dtype: int64

**Conclusiones**

Podemos observar que no tiene registros faltantes. 

**Categoria de trabajo**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?

**Análisis**


In [7]:
df["categoria_de_trabajo"].value_counts()

relacion_de_dependencia    22696
monotibutista               2541
empleado_municipal          2093
empleado_provincial         1298
responsable_inscripto       1116
empleadao_estatal            960
trabajo_voluntariado          14
sin_trabajo                    7
Name: categoria_de_trabajo, dtype: int64

In [8]:
# ¿Tiene valores nulos?
df["categoria_de_trabajo"].isnull().sum()

1836

**Conclusiones**

Podemos observar que faltan 1836 de los 32561 valores (5,6 %). 

**Barrio**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [11]:
# ¿Tiene valores nulos?
df["barrio"].isnull().sum()

583

In [9]:
df["barrio"].value_counts()

Palermo                29170
Belgrano                 643
San Isidro               198
Villa Urquiza            137
Recoleta                 121
La Boca                  114
Agronomia                106
Almagro                  100
Balvanera                 95
Puerto Madero             90
Caballito                 81
Boedo                     80
Barracas                  75
Chacarita                 73
Coghland                  70
Floresta                  67
Constitucion              64
Colegiales                62
Flores                    60
Liniers                   59
Monte Castro              51
Mataderos                 44
Nueva Pompeya             43
Monserrat                 37
nuñez                     34
Parque Chacabuco          31
Villa Luro                29
Parque Avellaneda         29
Parque Chas               28
La Paternal               24
Retiro                    20
Villa Devoto              19
Villa Soldati             19
Villa Real                18
San Telmo     

**Conclusiones**

Podemos observar que esta columna solo tiene un faltante de 583 sobre los 32561 valores (1,8 %). La mayoria de los registros corresponde al valor Palermo, los datos no estan balanceados.

**Edad**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [14]:
# ¿Tiene valores nulos?
df["edad"].isnull().sum()

0

In [15]:
df["edad"].value_counts()

36    898
31    888
34    886
23    877
35    876
     ... 
83      6
85      3
88      3
87      1
86      1
Name: edad, Length: 73, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos.

"Conclusion"

**Educación alcanzada**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [18]:
# ¿Tiene valores nulos?
df["educacion_alcanzada"].isnull().sum()

0

In [19]:
df["educacion_alcanzada"].value_counts()

4_anio                10501
5_anio                 7291
universidad_3_anio     5355
universidad_4_anio     1723
universidad_1_anio     1382
2_anio                 1175
universidad_2_anio     1067
1_anio                  933
7-8_grado               646
universiada_5_anio      576
9_grado                 514
3_anio                  433
universiada_6_anio      413
5-6_grado               333
1-4_grado               168
preescolar               51
Name: educacion_alcanzada, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos.

"Conclusion"

**Estado marital**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [20]:
# ¿Tiene valores nulos?
df["estado_marital"].isnull().sum()

0

In [21]:
df["estado_marital"].value_counts()

matrimonio_civil      14976
sin_matrimonio        10683
divorciado             4443
separado               1025
viudo_a                 993
pareja_no_presente      418
matrimonio_militar       23
Name: estado_marital, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos.

"Conclusion"

**Ganancia, perdida declarada bolsa argentina**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [22]:
# ¿Tiene valores nulos?
df["ganancia_perdida_declarada_bolsa_argentina"].isnull().sum()

0

In [23]:
df["ganancia_perdida_declarada_bolsa_argentina"].value_counts()

 0        28330
 15024      347
 7688       284
 7298       246
-1902       202
          ...  
 22040        1
 2538         1
-2467         1
-2163         1
-2080         1
Name: ganancia_perdida_declarada_bolsa_argentina, Length: 210, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos.

"Conclusion"

**Genero**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [24]:
# ¿Tiene valores nulos?
df["genero"].isnull().sum()

0

In [25]:
df["genero"].value_counts()

hombre    21790
mujer     10771
Name: genero, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos y solo dos valores posibles.

"Conclusion"

**Horas de trabajo registradas**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [26]:
# ¿Tiene valores nulos?
df["horas_trabajo_registradas"].isnull().sum()

0

In [27]:
df["horas_trabajo_registradas"].value_counts()

40    15217
50     2819
45     1824
60     1475
35     1297
      ...  
92        1
94        1
87        1
74        1
82        1
Name: horas_trabajo_registradas, Length: 94, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos. (Hay unos valores mas representativos)

"Conclusion"

**Religion**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [28]:
# ¿Tiene valores nulos?
df["religion"].isnull().sum()

0

In [29]:
df["religion"].value_counts()

cristianismo    27816
judaismo         3124
budismo          1039
ateismo           311
otro              271
Name: religion, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos. (Hay unos valores mas representativos)

"Conclusion"

**Rol familiar registrado**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [30]:
# ¿Tiene valores nulos?
df["rol_familiar_registrado"].isnull().sum()

0

In [31]:
df["rol_familiar_registrado"].value_counts()

casado         13193
sin_familia     8305
con_hijos       5068
soltero_a       3446
casada          1568
otro             981
Name: rol_familiar_registrado, dtype: int64

**Conclusiones**

Podemos observar que no tiene valores nulos. (Hay unos valores mas representativos y tambien hay que unificar los valores de casado y casada)

"Conclusion"

**Trabajo**

**Preguntas:**

* ¿La columna aporta información relevante?
* ¿El dato esta completo en todas las filas?
* ¿Los datos tienen mucha varianza?
* ¿Aporta valor para realizar una predicción?
* ¿Los datos estan balanceados?

**Análisis**

In [32]:
# ¿Tiene valores nulos?
df["trabajo"].isnull().sum()

1843

In [33]:
df["trabajo"].value_counts()

profesional_especializado    4140
reparador                    4099
directivo_gerente            4066
entretenimiento              3770
ventas                       3650
otros                        3295
inspector                    2002
transporte                   1597
limpiador                    1370
sector_primario               994
soporte_tecnico               928
seguridad                     649
servicio_domestico            149
ejercito                        9
Name: trabajo, dtype: int64

**Conclusiones**

Podemos observar que esta columna tiene un faltante de 1843 del total de valores (5,7 %). (Hay unos valores pocos representativos)

"Conclusion"

**Relación entre columnas**

**Conclusión final**

**Feature Engineering**

In [ ]:
def FeatureEngineering(df):
    
    df_clean = df.copy()
            
    return df_clean

In [ ]:
def Predict(df):
    pred = []
    values = df.values

    # for val in values:
    #     if val[1] == 'hombre':
    #         pred.append(0)
    #     else:
    #         if (val[0] == '4d' and val[4] == 'Palermo'):
    #             pred.append(0)
    #         else:
    #             pred.append(1)
     
    return pred

In [ ]:
def baseline(df):
    """    
    Input: dataframe
    Output: lista con las predicciones    
    """    
    # Aplicamos feature engineering a los datos de entrada
    entrada = FeatureEngineering(df)
    pred = Predict(entrada)
    
    return pred

In [ ]:
# Generamos dataframe para probar
df_datos_test = GetData()
df_datos_valor_adquisitivo = df_datos_test["tiene_alto_valor_adquisitivo"]

predicciones = baseline(df_datos_test)
accuracy_score(df_datos_valor_adquisitivo, predicciones)